src/User/Types.hs

```diff
@@ -14,13 +15,13 @@ data User = User
   , userBirthDate :: Day
   , userPassword  :: String
   , userGender    :: Gender
-  , userIncome    :: Double}
+  , userIncome    :: Scientific}
```

Comment:
Not sure about this type here... Why are you using Scientific notation to describe an income?
What about creating a new type to wrap all your money needs
```hs
newtype Currency = Currency  { unCurrency :: Double  } 
  deriving (Show, Eq, Num)
```

you can also derive from `PathPiece` to format the money when you need to use it in a page... for example you can do something like

```hs
instance PathPiece Currency where
  fromPathPiece = fmap Currency . readMaybe @Double . T.unpack
  toPathPiece = T.pack . printf "%.2f" . unCurrency
```